File: frontend/src/services/supplierService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  PageResponse, 
  SupplierResponse, 
  SupplierRequest, 
  TableParams 
} from '@/types';

export const supplierService = {
  getAll: (params?: TableParams): Promise<ApiResponse<PageResponse<SupplierResponse>>> => {
    return apiClient.get('/suppliers', { params });
  },

  getById: (id: number): Promise<ApiResponse<SupplierResponse>> => {
    return apiClient.get(`/suppliers/${id}`);
  },

  create: (data: SupplierRequest): Promise<ApiResponse<SupplierResponse>> => {
    return apiClient.post('/suppliers', data);
  },

  update: (id: number, data: SupplierRequest): Promise<ApiResponse<SupplierResponse>> => {
    return apiClient.put(`/suppliers/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/suppliers/${id}`);
  },

  exportExcel: (search?: string): Promise<Blob> => {
    return apiClient.get('/suppliers/export', {
      params: { search },
      responseType: 'blob',
    });
  },
};

File: backend/src/main/java/com/scim/service/SupplierService.java
package com.scim.service;

import com.scim.dto.request.SupplierRequest;
import com.scim.dto.response.PageResponse;
import com.scim.dto.response.SupplierResponse;
import org.springframework.data.domain.Pageable;

public interface SupplierService {

    PageResponse<SupplierResponse> getAll(String search, Pageable pageable);

    SupplierResponse getById(Long id);

    SupplierResponse create(SupplierRequest request);

    SupplierResponse update(Long id, SupplierRequest request);

    void delete(Long id);

    byte[] exportToExcel(String search);
}
